.github/workflows/autolabel_action.yaml

```diff
@@ -0,0 +1,55 @@
+name: Autolabel PRs
+
+on: pull_request
+
+jobs:
+  apply_one_t_label:
+    runs-on: ubuntu-latest
+    # Don't run on forks, where we wouldn't have permissions to add the label anyway.
+    if: github.repository == 'leanprover-community/mathlib4'
+    permissions:
+      issues: write
+      checks: write
+      pull-requests: write
+      contents: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: install elan
+        run: |
+          set -o pipefail
+          curl -sSfL https://github.com/leanprover/elan/releases/download/v3.1.1/elan-x86_64-unknown-linux-gnu.tar.gz | tar xz
+          ./elan-init -y --default-toolchain none
+          echo "$HOME/.elan/bin" >> "${GITHUB_PATH}"
+
+      - name: get cache for `Lean.Elab.Command`
+        run: |
+          git checkout master
+          git checkout -
+          lake exe cache get Mathlib/Tactic/Linter/HashCommandLinter.lean
+
+      - name: Auto-label script
+        run: |
+          printf '\nproduce_labels! "git"\n\nproduce_labels "git"\n' >> Mathlib/Tactic/Autolabels/Labels.lean
+          lake build Mathlib.Tactic.Autolabels.Labels
+          labels="$(lake build Mathlib.Tactic.Autolabels.Labels |
+            sed -n 's=.*Applicable labels: *\(.*\)=\1=p')"
```

Comment:
This approach feels a bit hacky.

I have a few questions about this step, to which I think the explanation could be added as comments here:

- the `printf >>` hack is to avoid the noise on building normally, right?
- what's the reason to do 'lake build' twice? Has that also to do with the output?
- do you even need `produce_labels!` or are you just `sed`ing for the output of `produce_labels`?